futures-util/src/try_stream/mod.rs

```diff
@@ -219,4 +222,41 @@ pub trait TryStreamExt: TryStream {
     {
         TryCollect::new(self)
     }
+
+    /// Attempt to filter the values produced by this stream while
+    /// simultaneously mapping them to a different type according to the
+    /// provided asynchronous closure.
+    ///
+    /// As values of this stream are made available, the provided function will
+    /// be run on them. If the future returned by the predicate `f` resolves to
+    /// [`Some(item)`](Some) then the stream will yield the value `item`, but if
+    /// it resolves to [`None`] then the next value will be produced.
+    ///
+    /// All errors are passed through without filtering in this combinator.
+    ///
+    /// Note that this function consumes the stream passed into it and returns a
+    /// wrapped version of it, similar to the existing `filter_map` methods in
+    /// the standard library.
+    ///
+    /// # Examples
+    /// ```
+    /// use futures::executor::block_on;
```

Comment:
Nit: we've been migrating the docs examples over to use `async`/`await` syntax, so rather than using two `block_on`s here, switch them to `await!`s and wrap the whole thing in a `block_on(async { .. })` as in [this example](https://github.com/rust-lang-nursery/futures-rs/blob/master/futures-util/src/try_stream/mod.rs#L40).